src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/GuidelinesPage.java

```diff
@@ -2,107 +2,141 @@
-            wait.forElementVisible(editModal);
+  public DiscussionsPage clickBackToDiscussions() {
+    wait.forElementClickable(backToDiscussionsButton);
+    backToDiscussionsButton.click();
 
-            return editModal.isDisplayed();
-        } catch (TimeoutException e) {
-            PageObjectLogging.logInfo("editModal is not displayed", e);
+    return new DiscussionsPage();
+  }
 
-            return false;
-        }
-    }
+  public void clickEditGuidelines() {
+    wait.forElementNotVisible(By.className("discussion-standalone-editor"));
+    editButton.click();
+  }
 
-    public boolean isGuidelinesHeroUnitDisplayed() {
-        try {
-            wait.forElementVisible(heroUnit);
+  public boolean isModalGuidelinesDisplayed() {
+    try {
+      wait.forElementVisible(editModal);
 
-            return heroUnit.isDisplayed();
-        } catch (TimeoutException e) {
-            PageObjectLogging.logInfo("heroUnit is not displayed", e);
+      return editModal.isDisplayed();
+    } catch (TimeoutException e) {
+      PageObjectLogging.logInfo("editModal is not displayed", e);
 
-            return false;
-        }
+      return false;
     }
+  }
 
-    public boolean isGuidelinesTextDisplayed() {
-        try {
-            wait.forElementVisible(contentText);
+  public boolean isGuidelinesHeroUnitDisplayed() {
+    return isSthg(heroUnit);
+  }
 
-            return contentText.isDisplayed();
-        } catch(TimeoutException e) {
-            PageObjectLogging.logInfo("contentText is not displayed", e);
+  private boolean isSthg(WebElement element) {
+    boolean result = false;
 
-            return false;
-        }
+    try {
+      wait.forElementVisible(element);
+      result = element.isDisplayed();
+    } catch (TimeoutException e) {
+      PageObjectLogging.logInfo("heroUnit is not displayed", e);
     }
 
-    public boolean isEditButtonDisplayed() {
-        try {
-            wait.forElementVisible(editButton);
-
-            return editButton.isDisplayed();
-        } catch(TimeoutException e) {
-            PageObjectLogging.logInfo("Edit button is not displayed", e);
-
-            return false;
-        }
+    return result;
+  }
+
+  public boolean isGuidelinesTextDisplayed() {
+    return isSthg(contentText);
+  }
+
+  public boolean isEditButtonDisplayed() {
+    return isSthg(editButton);
+  }
+
+  public void clickSaveButton() {
+    wait.forElementClickable(saveButton);
+    saveButton.click();
+  }
+
+  public void guidelinesAddNewText(String contentText) {
+    driver.findElement(By.className("discussion-standalone-editor-textarea")).clear();
+    driver.findElement(By.className("discussion-standalone-editor-textarea")).sendKeys(contentText);
+    clickSaveButton();
+  }
+
+  public void verifyContentNevTextInGuidelines(String content) {
+    wait.forTextInElement(contentText, content);
+  }
+
+  public void addNewTextToGuidelines() {
+    String guidelinesText = TextGenerator.createUniqueText();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "guidelinesText" which hides the field declared at line 42. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)